openstack/containerinfra/v1/certificates/requests.go

```diff
@@ -1,9 +1,28 @@
 package certificates
 
 import (
+	"net/http"
+
 	"github.com/gophercloud/gophercloud"
 )
 
+// CreateOptsBuilder allows extensions to add additional parameters
+// to the Create request.
+type CreateOptsBuilder interface {
+	ToCertificateCreateMap() (map[string]interface{}, error)
+}
+
+// CreateOpts represents options used to create a certificate.
+type CreateOpts struct {
+	BayUUID string `json:"bay_uuid" required:"true"`
```

Comment:
I tried an experiment by adding `ClusterUUID` as a field and, sure enough, it still worked. I think that's due to this line of code: 

https://github.com/openstack/magnum/blob/master/magnum/api/controllers/v1/certificate.py#L89

Given that, we can have the following in `CreateOpts`:

```go
ClusterUUID string `json:"cluster_uuid,omitempty" xor:"BayUUID"`
BayUUID     string `json:"bay_uuid,omitempty" xor:"ClusterUUID"`
```

`xor` is an internal Gophercloud tag that will require either one or the other.